common/source/docs/common-dock-mode.rst

```diff
@@ -0,0 +1,103 @@
+(`AC_PrecLand <https://github.com/ArduPilot/ardupilot/tree/master/libraries/AC_PrecLand>`__) to estimate the position of the docking target.
+
+Successful tests have been conducted using the OpenMV camera as a companion computer to perform auto-docking.
+However, any companion computer equipped with an RGB camera (like OpenMV camera, Raspberry Pi, Jetson Nano etc.) which can estimate pose
+of marker placed at docking target and send mavlink messages to the vehicle can be used.
+
+The OpenMV Camera
+=================
+
+OpenMV camera is a pocket sized camera board which is capable of performing various computer vision tasks.
+It can transmit mavlink messages over UART interface and hence it can be used as a companion computer to perform auto-docking.
+The camera can be purchased directly from `openmv.io <https://openmv.io/collections/cams>`__
+
+Connection to Autopilot
+=======================
+
+The companion computer needs to be connected to the autopilot through UART interface.
+
+.. image:: ../../../images/openmv-pixhawk.jpg
+   :target: ../_images/openmv-pixhawk.jpg
+   :width: 600px
+
+- Connect the camera’s P4, VIN and GND pins to one of the autopilot’s serial ports as shown above. In the image above the OpenMV cam is connected to a Pixhawk’s Telem2 port.
+  The connection scheme would be same for the other alternatives too. The user just needs to make a connection between
+  TX pin on the companion computer to RX pin of the telemetry port on the autopilot and power up the autopilot.
+- Set :ref:`SERIAL2_PROTOCOL <SERIAL2_PROTOCOL>` = 1 (MAVLink) if using Serial2/Telem2, if connected to another serial port use the corresponding SERIALx_PROTOCOL parameter
+- Set :ref:`SERIAL2_BAUD <SERIAL2_BAUD>` = 115
+
+Generating AprilTag Markers
+===========================
+
+.. image:: ../../../images/openmv-apriltag-generator.png
+   :target: ../_images/openmv-apriltag-generator.png
+   :width: 600px
+
+The OpenMV camera has capability to detect the AprilTags. The OpenMV IDE comes with built-in AprilTag generator. The steps to generate an AprilTag using the same are:
+
+- Go to Tools -> Machine Vision -> AprilTag generator -> TAG36H11 Family (587 Tags - Recommended). This will open a small dialog box.
+- Set the range of tags you want to generate. For eg. if you want to generate tags numbered 1 to 5, set Min to 1 and Max to 5.
+- Click ok and save the tags to the desired directory.
+
+Uploading Code to the Camera
+============================
+
+The OpenMV camera can be programmed using the OpenMV IDE. Detailed instructions to do so can be found `here <http://docs.openmv.io/openmvcam/tutorial/index.html>`__.
+
+- Connect the OpenMV camera to your PC using  a micro USB cable.
+- Open the OpenMV IDE and click the connect button at the bottom left corner of the IDE window.
+- Go to file menu, create new file and copy `this <https://github.com/shiv-tyagi/openmv-examples/blob/master/apriltag_autodock.py>`__ script to the editor window.
```

Comment:
this script should be moved into Tools/scripts in the ardupilot repo and linked from there rather than your personal repo

Reply:
I have updated the script at openmv directly and have linked it from their repository.